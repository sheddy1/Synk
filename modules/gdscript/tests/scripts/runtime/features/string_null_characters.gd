func test():
	print("\0" != null)
	print("\0" != "")
	print("\0" == "\u0000")
	print("\0" == String.chr(0))
	print("\0".length() == 1)

	var text := ""
	for n in 10:
		text += "\0"
	print(text.length() == 10)

	text = "string\0split\u0000by%snull" % String.chr(0)
	var split := text.split("\0")
	print(split)
	var join := "\u0000".join(split)
	print(join == text)

	var escape := text.c_escape()
	print(escape)
	join = escape.c_unescape()
	print(join == text)

	var buffer := text.to_ascii_buffer()
	print(buffer)
	join = buffer.get_string_from_ascii()
	print(join != text)

	buffer = text.to_utf8_buffer()
	print(buffer)
	join = buffer.get_string_from_utf8()
	print(join == text)

	buffer = text.to_utf16_buffer()
	print(buffer)
	join = buffer.get_string_from_utf16()
	print(join == text)

	buffer = text.to_utf32_buffer()
	print(buffer)
	join = buffer.get_string_from_utf32()
	print(join == text)

	buffer = [0x74, 0x72, 0x61, 0x69, 0x6c, 0, 0, 0, 0, 0] # "trail\0\0\0\0\0"
	print(buffer.get_string_from_ascii() == "trail")
	print(buffer.get_string_from_utf8() == "trail\0\0\0\0\0")
