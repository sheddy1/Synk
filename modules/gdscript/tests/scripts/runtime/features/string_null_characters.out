GDTEST_OK
true
true
true
true
true
true
["string", "split", "by", "null"]
true
string\0split\0by\0null
true
Unicode parsing error: ASCII treats NUL as termination, truncating result.
[115, 116, 114, 105, 110, 103]
true
[115, 116, 114, 105, 110, 103, 0, 115, 112, 108, 105, 116, 0, 98, 121, 0, 110, 117, 108, 108]
true
[115, 0, 116, 0, 114, 0, 105, 0, 110, 0, 103, 0, 0, 0, 115, 0, 112, 0, 108, 0, 105, 0, 116, 0, 0, 0, 98, 0, 121, 0, 0, 0, 110, 0, 117, 0, 108, 0, 108, 0]
true
[115, 0, 0, 0, 116, 0, 0, 0, 114, 0, 0, 0, 105, 0, 0, 0, 110, 0, 0, 0, 103, 0, 0, 0, 0, 0, 0, 0, 115, 0, 0, 0, 112, 0, 0, 0, 108, 0, 0, 0, 105, 0, 0, 0, 116, 0, 0, 0, 0, 0, 0, 0, 98, 0, 0, 0, 121, 0, 0, 0, 0, 0, 0, 0, 110, 0, 0, 0, 117, 0, 0, 0, 108, 0, 0, 0, 108, 0, 0, 0]
true
true
true
